src/runtime/CSRTensor.cpp

```diff
@@ -0,0 +1,272 @@
+        }
+    }
+
+    // This coordinates point to a zero value
+    return nullptr;
+}
+
+std::unique_ptr<ITensor> CSRTensor::to_dense()
+{
+    ARM_COMPUTE_ERROR_ON_MSG(info()->data_layout() != DataLayout::NCHW, "CSRTensor only supports NCHW layout at the moment");
+
+    auto tensor = std::make_unique<Tensor>();
+    tensor->allocator()->init(info()->clone()->set_tensor_format(TensorFormat::Dense));
+    tensor->allocator()->allocate();
+
+    const size_t first_elem_offset = info()->offset_first_element_in_bytes();
+    uint8_t                  *data = tensor->buffer() + first_elem_offset;
+    const size_t      element_size = info()->element_size();
+    const size_t        total_size = info()->total_size();
+    const uint8_t     *row_offsets = _allocator.data();
+    const uint8_t     *col_indices = _allocator.data() + _crow_bytes;
+    const uint8_t          *values = _allocator.data() + _crow_bytes + _col_bytes;
+    size_t                 element = 0;
+
+    std::memset(data, 0, total_size);
+
+    for(size_t i = 0, j = 1, row = 0; j < _crow_bytes / index_size; ++i, ++j, ++row)
+    {
+        const int32_t current_col = *reinterpret_cast<const int32_t *>(row_offsets + (i * index_size));
+        const int32_t next_col = *reinterpret_cast<const int32_t *>(row_offsets + (j * index_size));
+
+        if(current_col == next_col)
+        {
+            continue;
+        }
+
+        for(size_t current = static_cast<size_t>(current_col); current < static_cast<size_t>(next_col); ++current)
+        {
+            const size_t col = *reinterpret_cast<const int32_t *>(col_indices + (current * index_size));
+            const uint8_t *value_ptr = values + (element * element_size);
+
+            std::memcpy(data + (row * info()->dimension(1) + col) * element_size, value_ptr, element_size);
+            element++;
+        }
+    }
+
+    return tensor;
+}
+
+void CSRTensor::associate_memory_group(IMemoryGroup *memory_group)
+{
+    _allocator.set_associated_memory_group(memory_group);
+}
+
+#ifdef ARM_COMPUTE_ASSERTS_ENABLED
+void CSRTensor::print(std::ostream &os) const
+{
+    const uint8_t *_row_offsets = _allocator.data();
+    const uint8_t *_col_indices = _allocator.data() + _crow_bytes;
+    const uint8_t       *values = _allocator.data() + _crow_bytes + _col_bytes;
+
+    os << "r_offsets: [";
+    for(size_t i = 0; i < _crow_bytes / index_size; ++i)
+    {
+        os << *reinterpret_cast<const int32_t *>(_row_offsets + (i * index_size));
+        if (i < _crow_bytes / index_size - 1)
+        {
+            os << ", ";
+        }
+    }
+    os << "] cols: [";
+    for(size_t i = 0; i < _col_bytes / index_size; ++i)
+    {
+        os << *reinterpret_cast<const int32_t *>(_col_indices + (i * index_size));
+        if (i < _col_bytes / index_size - 1)
+        {
+            os << ", ";
+        }
+    }
+    os << "] values: ";
+    print_values(os, values, 0, nnz());
+}
+#endif // ARM_COMPUTE_ASSERTS_ENABLED
```

Comment:
Move this to existing `void ITensor::print(std::ostream &s, IOFormatInfo io_fmt) const`